src/task_scl_eff_pot_hab.py

```diff
@@ -3,180 +3,200 @@
-            geometryInNativeProjection = True
-            ) \
-            .select(['area_sum', 'hii_mode', 'label', 'max', 'patch_mode', 'stpStn_mode'], \
-                ['patch_area', 'hii', 'label', 'prob', 'min_patch_area', 'min_stp_stn_area'])
+        self.structural_habitat, _ = self.get_most_recent_image(
+             ee.ImageCollection(self.inputs["structural_habitat"]["ee_path"])
+        )
+        self.probability, _ = self.get_most_recent_image(
+             ee.ImageCollection(self.inputs["probability"]["ee_path"])
+        )
+        self.hii, _ = self.get_most_recent_image(
+             ee.ImageCollection(self.inputs["hii"]["ee_path"])
+        )
+        self.historic_range = ee.Image(self.inputs["historic_range"]["ee_path"])
+        self.extirpated = ee.Image(self.inputs["extirpated"]["ee_path"])
+        self.elevation = ee.Image(self.inputs["elevation"]["ee_path"])
+        self.ecoregion_country = ee.Image(self.inputs["ecoregion_country"]["ee_path"])
+        self.density = ee.Image(self.inputs["density"]["ee_path"])
+        self.water = ee.Image(self.inputs["water"]["ee_path"])
+        self.set_aoi_from_ee(
+            "{}/{}/sumatra_poc_aoi".format(self.ee_rootdir, self.species)
+        )
 
-        return hii_grp_poly.first()
-        # blob = "ls_stats/{}/{}/{}".format(self.species, self.taskdate, landscape_key)
-        # self.export_fc_ee(ls_countries_biomes_pas, "scl-pipeline", blob)
+    def calc(self):
+        elev_mask = self.elevation.lt(self.thresholds["elev"]).selfMask()
+        str_hab_mask = self.structural_habitat.gt(self.thresholds["str_lc"]).selfMask()
+        str_hab = str_hab_mask.updateMask(elev_mask).updateMask(self.water)
+        str_hab_patch = (
+            str_hab.connectedPixelCount(16, True)
+            .gte(self.thresholds["patch_size"])
+            .selfMask()
+            .reproject(crs="EPSG:4326", scale=self.scale)
+        )
+        current_range = (
+            self.extirpated.updateMask(self.historic_range.eq(0)).eq(1).selfMask()
+        )
 
-foo = SCLPolygons.delineate_polygons(SCLPolygons)
-print(foo.getInfo())
+        connected_structural_habitat = str_hab_patch.updateMask(current_range)
+        potential_habitat = connected_structural_habitat.updateMask(
+            self.hii.lt(self.thresholds["hii"])
+        ).selfMask()
+        excluded_habitat = connected_structural_habitat.updateMask(
+            self.hii.gte(self.thresholds["hii"])
+        ).selfMask()
+        effective_potential_habitat = potential_habitat.addBands(
+            excluded_habitat
+        ).rename(["eff_pot_hab", "excl_hab"])
+
+        # probability_calc = self.elevation.expression(
+        #     "1 - (2.71828 ** (-(2.71828 ** (- 0.069 - 0.044 * ((dem - 639) / 592) - 0.12 * ((hii - 7.72) / 5.47)))))",
+        #     {"dem": self.elevation, "hii": self.hii},
+        # )
+
+        high_probability = self.probability.gte(self.thresholds["probability"]).multiply(
```

Comment:
I still don't really understand the `multiply` calls -- why multiply by 2, or even more why multiply by 1? Maybe you can indicate in a comment?

Reply:
we want a single classified image of high and low probability. there are a number of means to get this.  we could use a image.where (which might be more clean). This was very quick/easy to code up.  image.gte creates a 0/1 binary image. same for image.lt.  If we just combine these two, we end up with an image where both high and low prob is classified as 1. the multiply by 1 is completely unnecessary. could also use a image.remap function.